### PR TITLE
Feature: Union schema compatibility

### LINKED ISSUE #46: [Feature] Add union schema capability
*Copied from [fivetran/dbt_facebook_ads_source #29](https://github.com/fivetran/dbt_facebook_ads_source/issues/29).*

### Is there an existing feature request for this?

- [X] I have searched the existing issues

### Describe the Feature

Issue for adding the union schema capability.

### Describe alternatives you've considered

_No response_

### Are you interested in contributing this feature?

- [ ] Yes.
- [ ] Yes, but I will need assistance and will schedule time during your [office hours](https://calendly.com/fivetran-solutions-team/fivetran-solutions-team-office-hours) for guidance.
- [ ] No.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by fivetran-catfritz**
Issue: #46 

Confirm the following files were correctly updated automatically:
- [x] CHANGELOG
- [x] README
- [x] stg_apple_search_ads.yml, src_apple_search_ads.yml, apple_search_ads.yml (depending if source or transform)
- [x] docs.md
- [ ] ~joins~
- [x] window functions (partition by)
- [x] 'source_relation' column added in staging and transform models
- [x] tests

Manual updates:
- [x] Add `source_relation` to downstream models if necessary
- [x] Finish any incomplete/incorrect joins/partitions
- [x] Update tests to include `source_relation` in unique-combination-of-cols if necessary
    - May need to remove some uniqueness tests in favor of the combo test
    
Validation:
- [x] `dbt run` locally passes
- [x] `dbt test` locally passes

### CODE REVIEW

**Comment by fivetran-catfritz** [comment]
Note--we need to decide if we update identifier for `ad_level_report` table or update `fivetran_utils.union_data` to handle divergent naming.

It technically passes buildkite, but reviewing the logs, it can't find a lot of the seed tables because of the identifier issues, which would mean problems for others using identifiers.

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md]: Actually, can we remove the README "hide" sections in these updates. This way we are able to link directly to the section in the README where the users needs to understand how to configure the union feature.

Otherwise, I worry this cool feature is hidden from people actually using it.

File: packages.yml
Comment (fivetran-joemarkiewicz) [inline packages.yml]: Reminder to update before merge and release

**Review by fivetran-joemarkiewicz — APPROVED** [review]
Only a few small comments to make minor updates to the README and packages.yml before merging.

Again let's hold off to release until we can do them all together. Otherwise, this looks good to go!

File: CHANGELOG.md
Comment (fivetran-catfritz) [reply CHANGELOG.md]: Updated.

**Comment by fivetran-catfritz** [comment]
Merging for release on 12-Oct-2023.

**Pull request merged**
